crypto/async/arch/async_null.c

```diff
@@ -7,10 +7,12 @@
  * https://www.openssl.org/source/license.html
  */
 
-/* This must be the first #include file */
-#include "../async_locl.h"
+#include <openssl/opensslconf.h>
 
 #ifdef ASYNC_NULL
```

Comment:
The essential issue with this PR is this...  `ASYNC_NULL` is defined by `async_null.h`, which is included *last* (because it's a fallback if `async_posix.h` and `async_win.h` don't define `ASYNCH_ARCH`) by `../async_locl.h`.  Now, you don't include `async_null.h` at all from this module, which means that `ASYNC_NULL` will *never* be defined here, which means that build will fail on machines that aren't POSIXsly or Windowsy enough, because *all* the C files here will generate empty object files.

Reply:
Exactly.  So it should move to opensslconf.h.in.

Reply:
So what you're saying is that you really want to redesign entirely how this autodetection is done, and move it all to Configure.  In that case, this PR is seriously incomplete.  Also, Configure doesn't support that kind of detection very well...

Reply:
No, I don't want auto-detection.  Just like we can say no-threads we should be able to say no-async

Reply:
FYI: "no-async" is already supported.

Reply:
yes, but the OPENSSL_NO_ASYNC define isn't created anywhere.  It should be in opensslconf.h

Reply:
Errr, yes it is:
````
$ grep OPENSSL_NO_ASYNC opensslconf.h
#ifndef OPENSSL_NO_ASYNC
# define OPENSSL_NO_ASYNC
````